cloud/team/team.go

```diff
@@ -0,0 +1,708 @@
+package team
+
+import (
+	httpContext "context"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+
+	astrocore "github.com/astronomer/astro-cli/astro-client-core"
+	"github.com/astronomer/astro-cli/cloud/user"
+	"github.com/astronomer/astro-cli/context"
+	"github.com/astronomer/astro-cli/pkg/input"
+	"github.com/astronomer/astro-cli/pkg/printutil"
+)
+
+var (
+	errInvalidTeamKey           = errors.New("invalid team selection")
+	ErrInvalidTeamMemberKey     = errors.New("invalid team member selection")
+	ErrInvalidName              = errors.New("no name provided for the team. Retry with a valid name")
+	ErrTeamNotFound             = errors.New("no team was found for the ID you provided")
+	ErrWrongEnforceInput        = errors.New("the input to the `--enforce-cicd` flag")
+	ErrNoShortName              = errors.New("cannot retrieve organization short name from context")
+	ErrNoTeamsFoundInOrg        = errors.New("no teams found in your organization")
+	ErrNoTeamsFoundInWorkspace  = errors.New("no teams found in your workspace")
+	ErrNoTeamMembersFoundInTeam = errors.New("no team members found in team")
+	ErrNoUsersFoundInOrg        = errors.New("no users found in your organization")
+	teamPagnationLimit          = 100
+)
+
+func CreateTeam(name, description string, out io.Writer, client astrocore.CoreClient) error {
+	if name == "" {
+		return ErrInvalidName
+	}
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+	if ctx.OrganizationShortName == "" {
+		return user.ErrNoShortName
+	}
+	teamCreateRequest := astrocore.CreateTeamJSONRequestBody{
+		Description: &description,
+		Name:        name,
+	}
+	resp, err := client.CreateTeamWithResponse(httpContext.Background(), ctx.OrganizationShortName, teamCreateRequest)
+	if err != nil {
+		return err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(out, "Astro Team %s was successfully created\n", name)
+	return nil
+}
+
+func UpdateWorkspaceTeamRole(id, role, workspace string, out io.Writer, client astrocore.CoreClient) error {
+	err := user.IsWorkspaceRoleValid(role)
+	if err != nil {
+		return err
+	}
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+	if ctx.OrganizationShortName == "" {
+		return user.ErrNoShortName
+	}
+	if workspace == "" {
+		workspace = ctx.Workspace
+	}
+
+	teams, err := GetWorkspaceTeams(client, workspace, teamPagnationLimit)
```

Comment:
I wish there was get workspace team API by passing the team ID as well so that we don't need to iterate all teams, and also it would lead to less resource consumption on core API and DB 🙂

Reply:
Just added a GetTeam and GetUser method and if the id's are passed in I use that method instead of the list